### PR TITLE
skip test_trace_c10_ops if _caffe2 is not built

### LINKED ISSUE #18142: export BUILD_CAFFE2_OPS=0 breaks jit tests
## 🐛 Bug
```
ERROR: test_trace_c10_ops (__main__.TestJit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_jit.py", line 1431, in test_trace_c10_ops
    traced_model = torch.jit.trace(model, inputs)
  File "/data/users/zdevito/pytorch/torch/jit/__init__.py", line 668, in trace
    var_lookup_fn, _force_outplace)
  File "/data/users/zdevito/pytorch/torch/nn/modules/module.py", line 489, in __call__
    result = self._slow_forward(*input, **kwargs)
  File "/data/users/zdevito/pytorch/torch/nn/modules/module.py", line 479, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "test/test_jit.py", line 1414, in forward
    a, b = torch.ops._caffe2.GenerateProposals(
  File "/data/users/zdevito/pytorch/torch/_ops.py", line 60, in __getattr__
    op = torch._C._jit_get_operation(qualified_op_name)
RuntimeError: No such operator _caffe2::GenerateProposals
```

## To Reproduce

Built with

```
export BUILD_CAFFE2_OPS=0
```

## Expected behavior

Test is disabled or arranged not to fail if CAFFE2_OPS were not built.

### PR DESCRIPTION
**Pull request by houseroad**
fix https://github.com/pytorch/pytorch/issues/18142

### CODE REVIEW

**Review by smessmer — CHANGES_REQUESTED** [review]
Can we use `caffe2.core._RegisteredOperators` instead? If we check `torch.ops`, it'll not catch cases where the caffe2 operator library is loaded but exposing the operator to c10 got broken.

**Comment by houseroad** [comment]
Then in this test, we will import caffe2.python.core in test_jit.py (of course, only in this test case), if we are OK with that, then I can go with that.

File: test/test_jit.py [hunk 1]
```diff
@@ -1727,6 +1727,11 @@ def forward(self, x):
         FileCheck().check("mul").check("add").run(str(trace))
 
     def test_trace_c10_ops(self):
```
Comment (dzhulgakov) [inline test/test_jit.py]: would something like `@skipIf(hasattr(torch.ops._caffe2, 'GenerateProposals')` here work instead? it's more idiomatic

Comment (houseroad) [reply test/test_jit.py]: I tried that... actually, we did some magic in __set_state__/__get_state__, hasattr(torch.ops._caffe2, 'GenerateProposals') always return false...

Comment (dzhulgakov) [reply test/test_jit.py]: It'd be nice to fix hasattr to work (probably because we throw RuntimeError instead of AttributeError), but it's non-blocking

**Review by dzhulgakov — APPROVED** [review]

**Review by smessmer — APPROVED** [review]

**Review by zdevito — APPROVED** [review]
Ship it!

**Pull request closed without merging**